_posts/2021-01-23-intermediate-representation-for-quantum-computing.md

```diff
@@ -0,0 +1,39 @@
+
+# What is QIR?
+
+QIR is based on LLVM, which is a classical, well-spread, open-source intermediate representation. QIR is hardware-agnostic, which means that there are no specifications regarding the quantum platform, and even if now it is mainly used for representing Q# programs, it has the potential to represent any other programming language such as Qiskit.
+ 
+The fact that LLVM is the underlying language means that QIR is naturally able to process both classical and quantum logic. This feature is essential for the so-called hybrid algorithms, which have experienced an incredible boost in the last years. This kind of algorithms include both quantum and classical operations that influence each other. This is the case, for instance, of variational algorithms, where a parameter-dependent quantum circuit is run, followed by a classical post-processing and optimization part, which is later used to update the circuit in the quantum device. Part of the reason why these algorithms have generated such a great interest lately is their ability to tackle optimization problems – take for instance the Quantum Approximate Optimization Algorithm (QAOA), which can be used for solving graph problems such as Max-Cut, among other applications – or even quantum chemistry problems – for example, by using the Variational Quantum Eigensolver to approximate a Hamiltonian’s spectrum.
+
+![QIR workflow](/assets/images/qir_diagram.png)
+
+# Why is QIR interesting?
+ 
+We’ve already mentioned that QIR is an essential tool when running quantum algorithms on real hardware. But even if we do not want to go that far and are just developing algorithms at a more theoretical level, intermediate representations can play an important role! 
+
+First, there are optimization steps that can be performed at the intermediate level that can make the overall implementation more efficient. Investigating this optimisation of your input code can help to get a better understanding of where we can make algorithms more efficient and how to improve the quantum programming languages. 
+
+Moreover, you could use the intermediate representation to generate code that is later on given as input into a quantum simulator -- instead of a real device -- which could potentially use a different language than your source one. IThis way, we can easily compare and benchmark different languages and simulators using a common framework.
```

Comment:
```suggestion
Moreover, you could use the intermediate representation to generate code that is later on given as input into a quantum simulator -- instead of a real device -- which could potentially use a different language than your source one. In this way, we can easily compare and benchmark different languages or simulators using a common framework.
```